From 8a9a2c47c5885002b9bdf3ee0b54a5003bc166bb Mon Sep 17 00:00:00 2001
From: caiyinyu <caiyinyu@loongson.cn>
Date: Sat, 23 Jan 2021 14:35:59 +0800
Subject: [PATCH 1/2] Subject: [PATCH 1/1] Add PF_XDP, AF_XDP and SOL_XDP from
 Linux 4.18 to  bits/socket.h.

This patch adds the PF_XDP, AF_XDP and SOL_XDP macros from Linux 4.18 to
sysdeps/unix/sysv/linux/bits/socket.h.

    * sysdeps/unix/sysv/linux/bits/socket.h (PF_MAX): Set to 45.
    (PF_XDP): New macro.
    (AF_XDP): New macro.
    (SOL_XDP): New macro.

Change-Id: Ide7d3053081be9b2ebb751ce49030ab071b34f83
---
 sysdeps/unix/sysv/linux/bits/socket.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sysdeps/unix/sysv/linux/bits/socket.h b/sysdeps/unix/sysv/linux/bits/socket.h
index fa409f0fab..c3fbb21102 100644
--- a/sysdeps/unix/sysv/linux/bits/socket.h
+++ b/sysdeps/unix/sysv/linux/bits/socket.h
@@ -85,7 +85,8 @@ typedef __socklen_t socklen_t;
 #define PF_KCM		41	/* Kernel Connection Multiplexor.  */
 #define PF_QIPCRTR	42	/* Qualcomm IPC Router.  */
 #define PF_SMC		43	/* SMC sockets.  */
-#define PF_MAX		44	/* For now..  */
+#define PF_XDP		44	/* XDP sockets.  */
+#define PF_MAX		45	/* For now..  */
 
 /* Address families.  */
 #define AF_UNSPEC	PF_UNSPEC
@@ -135,6 +136,7 @@ typedef __socklen_t socklen_t;
 #define AF_KCM		PF_KCM
 #define AF_QIPCRTR	PF_QIPCRTR
 #define AF_SMC		PF_SMC
+#define AF_XDP		PF_XDP
 #define AF_MAX		PF_MAX
 
 /* Socket level values.  Others are defined in the appropriate headers.
@@ -164,6 +166,7 @@ typedef __socklen_t socklen_t;
 #define SOL_NFC		280
 #define SOL_KCM		281
 #define SOL_TLS		282
+#define SOL_XDP		283
 
 /* Maximum queue length specifiable by listen.  */
 #define SOMAXCONN	128
-- 
2.20.1

